### PR TITLE
adding nhc health checks for ndv4

### PR DESCRIPTION
**Pull request by rafsalas19**

### CODE REVIEW

File: ubuntu/common/install_health_checks.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+pushd /opt/azurehpc/test/
+
+git clone https://github.com/Azure/azurehpc-health-checks.git --branch $AZHC_VERSION
+
+pushd azurehpc-health-checks
+
+NHC_VERSION=1.4.3
+echo "Installed NHC verison $NHC_VERSION"
+
+wget -O nhc-$NHC_VERSION.tar.xz https://github.com/mej/nhc/releases/download/1.4.3/lbnl-nhc-1.4.3.tar.xz
```
Comment (abhamidipati0614) [inline ubuntu/common/install_health_checks.sh:18]: Maybe you can use the variable NHC_VERSION here in the url

File: ubuntu/common/install_health_checks.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+echo -e "\n"
+sudo make install
+popd
+
+echo -e "\nRunning set up script for custom tests"
+pushd customTests/
+./custom-test-setup.sh
+popd
+popd
+
```
Comment (abhamidipati0614) [inline ubuntu/common/install_health_checks.sh:34]: minor: pushd/ popd doesn't match

File: ubuntu/common/install_health_checks.sh (hunk 1 above)
Comment (jeseszhang1010) [inline ubuntu/common/install_health_checks.sh:18]: NIT: pls use ${NHC_VERSION} in 1.4.3/lbnl-nhc-1.4.3.tar.xz

Comment (rafsalas19) [reply ubuntu/common/install_health_checks.sh:18]: will address

File: ubuntu/common/install_health_checks.sh (hunk 2 above)
Comment (rafsalas19) [reply ubuntu/common/install_health_checks.sh:34]: let me address

File: ubuntu/common/install_health_checks.sh (hunk 1 above)
Comment (rafsalas19) [reply ubuntu/common/install_health_checks.sh:18]: will address

Comment (rafsalas19) [reply ubuntu/common/install_health_checks.sh]: addressed

Comment (rafsalas19) [reply ubuntu/common/install_health_checks.sh]: addressed

File: ubuntu/common/install_health_checks.sh (hunk 2 above)
Comment (rafsalas19) [reply ubuntu/common/install_health_checks.sh:33]: addressed

File: ubuntu/common/install_health_checks.sh (hunk 1 above)
Comment (abhamidipati0614) [reply ubuntu/common/install_health_checks.sh]: Maybe this one wget -O nhc-$NHC_VERSION.tar.xz https://github.com/mej/nhc/releases/download/1.4.3/lbnl-nhc-1.4.3.tar.xz to
wget -O nhc-$NHC_VERSION.tar.xz https://github.com/mej/nhc/releases/download/$NHC_VERSION/lbnl-nhc-$NHC_VERSION.tar.xz

File: ubuntu/common/install_health_checks.sh (hunk 2 above)
Comment (abhamidipati0614) [reply ubuntu/common/install_health_checks.sh:33]: I see 2 extra pushd now

File: ubuntu/ubuntu-22.x/ubuntu-22.04-hpc/install.sh [hunk 3]
```diff
@@ -52,6 +52,8 @@ $COMMON_DIR/copy_test_file.sh
 # install monitor tools
 $UBUNTU_COMMON_DIR/install_monitoring_tools.sh
 
+# install Azure/NHC Health Checks
+$UBUNTU_COMMON_DIR/install_health_checks.sh
```
Comment (jithinjosepkl) [inline ubuntu/ubuntu-22.x/ubuntu-22.04-hpc/install.sh:56]: How about Alma-Linux?

Comment (rafsalas19) [reply ubuntu/ubuntu-22.x/ubuntu-22.04-hpc/install.sh:56]: let me test then add

File: alma/common/install_health_checks.sh [hunk 4]
```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+
+set -e
+
+AZHC_VERSION=v0.2.0
+AOCC_VERSION=4.0.0-1
+
+# install dependency
+wget https://download.amd.com/developer/eula/aocc-compiler/aocc-compiler-${AOCC_VERSION}.x86_64.rpm
```
Comment (jithinjosepkl) [inline alma/common/install_health_checks.sh:10]: please move this outside of health checks install step.

File: ubuntu/ubuntu-22.x/ubuntu-22.04-hpc/install.sh (hunk 3 above)
Comment (rafsalas19) [reply ubuntu/ubuntu-22.x/ubuntu-22.04-hpc/install.sh:56]: addressed

File: alma/alma-8.x/alma-8.6-hpc/install.sh [hunk 5]
```diff
@@ -52,6 +52,9 @@ $COMMON_DIR/install_hpcdiag.sh
 #install monitoring tools
 $COMMON_DIR/../alma/common/install_monitoring_tools.sh
 
+# install Azure/NHC Health Checks
+$UBUNTU_COMMON_DIR/install_health_checks.sh
```
Comment (abhamidipati0614) [inline alma/alma-8.x/alma-8.6-hpc/install.sh:56]: Can you please also add this to the 8.7 tree?

Comment (rafsalas19) [reply alma/alma-8.x/alma-8.6-hpc/install.sh]: addressed

**Comment by abhamidipati0614** [comment]
@rafsalas19 the branch seems out of date. Pls update.

File: ubuntu/common/install_health_checks.sh [hunk 6]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (rafsalas19) [reply ubuntu/common/install_health_checks.sh:1]: So the there are slight differences in package manager dnf vs apt-get. Also in how the version number is annotated for AMD libs. If we unify the scripts I will have to put in an OS check. Which isn't an issue but that was my original reason why I split into 2 sets of files. Let me know how you want me to proceed?

[the remark this answers is not in the document]
Comment (jithinjosepkl) [reply ubuntu/common/install_health_checks.sh:1]: Oh, I am referring to the health-checks installation, not the AOCC installation.

[the remark this answers is not in the document]
Comment (rafsalas19) [reply ubuntu/common/install_health_checks.sh:1]: This makes sense. I forgot I handled the OS based dependencies in the NHC install file. Let me address.

[the remark this answers is not in the document]
Comment (rafsalas19) [reply ubuntu/common/install_health_checks.sh]: addressed

**Review by abhamidipati0614 — APPROVED** [review]

**Pull request merged**